core/src/test/java/org/apache/accumulo/core/client/lexicoder/BigDecimalLexicoderTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.lexicoder;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+
+import org.apache.accumulo.core.clientImpl.lexicoder.AbstractLexicoderTest;
+import org.junit.jupiter.api.Test;
+
+public class BigDecimalLexicoderTest extends AbstractLexicoderTest {
+
+  @Test
+  public void testSortOrder() {
+
+    assertSortOrder(new BigDecimalLexicoder(), BigDecimal::compareTo,
```

Comment:
If we don't pass the comparator wouldn't it still use the default comparator which would still not give us want we want?  Shouldn't we write our Comparator? Because our encoder and decoder know how data is stored and would compare it correctly.

Reply:
The concern I have isn't whether or not we're passing the comparator. The concern I have is that BigDecimal's own comparator does not guarantee a particular ordering of distinct input. It's non-deterministic, because it treats some values as equal when they differ in scale.